docs/content/envoy-primer.md

```diff
@@ -0,0 +1,347 @@
+---
+title: Policy Primer via Examples
+kind: envoy
+weight: 2
+---
+
+This page covers how to write policies for the content of the requests that are passed to OPA by Envoy's
+[External Authorization
+filter](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ext_authz_filter.html).
+
+## Writing Policies
+
+Let's start with an example policy that restricts access to an endpoint based on a user's role and permissions.
+
+```live:bool_example:module:read_only
+package envoy.authz
+
+import input.attributes.request.http as http_request
+
+default allow = false
+
+token = {"valid": valid, "payload": payload} {
+    [_, encoded] := split(http_request.headers.authorization, " ")
+    [valid, _, payload] := io.jwt.decode_verify(encoded, {"secret": "secret"})
+}
+
+allow {
+    is_token_valid
+    action_allowed
+}
+
+is_token_valid {
+  token.valid
+  now := time.now_ns() / 1000000000
+  token.payload.nbf <= now
+  now < token.payload.exp
+}
+
+action_allowed {
+  http_request.method == "GET"
+  token.payload.role == "guest"
+  glob.match("/people*", [], http_request.path)
+}
+
+action_allowed {
+  http_request.method == "GET"
+  token.payload.role == "admin"
+  glob.match("/people*", [], http_request.path)
+}
+
+action_allowed {
+  http_request.method == "POST"
+  token.payload.role == "admin"
+  glob.match("/people", [], http_request.path)
+  lower(input.parsed_body.firstname) != base64url.decode(token.payload.sub)
+}
+```
+
+The first line `package envoy.authz` declaration gives the (hierarchical) name `envoy.authz` to the rules in the
+remainder of the policy. If the OPA-Envoy [configuration](envoy-introduction#configuration) does not specify the `path`
+field, `envoy/authz/allow` will be considered as the default policy decision path. `data.envoy.authz.allow` will be the
+name of the policy decision to query in the default case.
+
+The line `import input.attributes.request.http as http_request`, defines a dependency on `attributes.request.http`
+which is part of the [input document](#input-document). The line also uses the `as` keyword so that `http_request` can
+be used as an alias for `input.attributes.request.http` in the rest of the policy.
```

Comment:
If the `as` keyword was not used, the policy would refer to `http` instead of the entire path. This sentence is slightly inaccurate.

Reply:
Removed use of `as` keyword as it doesn't seem necessary.